drupal/main.tf

```diff
@@ -0,0 +1,124 @@
+locals {
+  tags         = setunion(["terraform-cloudgov-managed"], var.tags)
+  s3_name      = "${var.name}-storage"
+  rds_name     = "${var.name}-database"
+  secrets_name = "${var.name}-secrets"
+  drupal_route = coalesce(var.route, "${var.name}.app.cloud.gov")
+}
+
+module "database" {
```

Comment:
Maybe consider conditional creation/dependency injection for DB and S3 here in case the consumer of the module needs to do things with an existing resource not owned by this module. Did I hear correctly that dependency injection enabled better mocking for the SpiffWorkflow module, and if so, would that help here too?

Reply:
Do you have an example of conditional creation working? I've wanted to do that in the past but I haven't been able to correctly set it up in a way that terraform was happy with. So I've been left with the choice of "does this module always handle this resource or does the user always have to pass it in" In _most_ cases I'd prefer "this module will create this resource unless this var is set"

Reply:
I did it in the SMTP broker to implement [the "domain/no-domain" features](https://github.com/GSA-TTS/datagov-brokerpak-smtp/tree/main?tab=readme-ov-file#featurescomponents).

- [Based on the variables, determine if the module should handle required resource creation](https://github.com/GSA-TTS/datagov-brokerpak-smtp/blob/dc070fc5aaa709eb09a4a21ed10fba9ce5566946/terraform/provision/main.tf#L4)
- [Use that information with `count` to toggle the creation of those resources](https://github.com/GSA-TTS/datagov-brokerpak-smtp/blob/main/terraform/provision/verification.tf)

Reply:
ohhhh, you know what - I think it would work since we're using names and not ids. I believe what won't work is:

```
locals {
  db_id = (var.db_id == "" ? cloudfoundry_service_instance.db.0.id : var.db_id)
}
```

because the `cloudfoundry_service_instance.db` doesn't get created, but since we're just operating on the names we can avoid referencing the conditionally-created instance. Though now I'm worried about how to prevent the app from being created before the DB is ready

Reply:
Yeah, I don't think you can make a `depends_on` refer to the exact ID of the optionally-created thing. It _might_ still work if you have the `depends_on` refer to `cloudfoundry_service_instance.db` array as a whole, since it'll still exist and just be zero-length. Not sure, you'd have to try it. 🤔